pydantic/type_adapter.py

```diff
@@ -227,6 +233,17 @@ def __init__(
             force=False,
         )
 
+    def _fetch_parent_frame(self, depth: int) -> FrameType | None:
+        frame = sys._getframe(depth)
```

Comment:
Because `parent_frame_namespace` returns the locals directly, but I need the globals as well to check for the module of the frame. Previously, we did another call to `sys._getframe` to fetch the globals (resulting in the frame being fetched twice), but it ends up being cleaner fetching the relevant frame once, and pulling up `f_locals` and `f_globals` from it.